### ISSUE TITLE
[PWM] Error thrown when set pulsewidth before period

### ISSUE THREAD

**Issue by cuiyanx** [01org/zephyr.js#359]
```
var pinA = pwm.open({ channel: pins.IO3 });
pinA.setPeriod(1000);
pinA.setPulseWidth(300);
```

It work well.

```
pinA.setPulseWidth(300);
pinA.setPeriod(1000);
```

**error message:**

`zjs_pwm_set: pulseWidth was greater than period`

**Comment by grgustaf** [comment]
My conclusion looking at the Zephyr API was it's not good to set these things separately because you're in an invalid state in between. They agreed and changed their API based on my suggestions. So hopefully we can follow suit. @zolkis may be interested, I am afraid I'm not digging in at the moment to see what it's like in the new APIs currently.

**Comment by grgustaf** [comment]
This is addressed by #928. Now the APIs require you to set period and pulseWidth at the same time. If pulseWidth is greater than period, they throw an error immediately.

**Comment by qiaojingx** [comment]
Verified with commit  1c67773 + #928. This issue is not reproducible.
Will close it after patch merged.
